engine/src/main/java/org/hibernate/search/engine/environment/classpath/spi/DefaultResourceResolver.java

```diff
@@ -42,7 +42,7 @@ public InputStream locateResourceStream(String name) {
 
 		if ( stripped != null ) {
 			try {
-				return new URL( stripped ).openStream();
+				return new URI( stripped ).toURL().openStream();
```

Comment:
Can you please check this will not change the behavior when "name" contains a URL escape such as `%20`? I know it's unlikely, but I don't want to mess with this rather... arcane code.

Reply:
So it seems that escaped characters won't cause a problem, but not escaped, on the other hand, would end up throwing an exception rather than getting escaped as before. I have tried it with the next test:
```java
Path path = folder.getRoot().toPath();

Path path1 = Files.createFile( path.resolve( "file%201" ) );
Path path2 = Files.createFile( path.resolve( "file 1" ) );

Path path = folder.getRoot().toPath();

Path path1 = Files.createFile( path.resolve( "file%201" ) );
Path path2 = Files.createFile( path.resolve( "file 1" ) );

try ( FileWriter fileWriter = new FileWriter( path1.toFile() ) ) {
	fileWriter.write( "file 1" );
}
try ( FileWriter fileWriter = new FileWriter( path2.toFile() ) ) {
	fileWriter.write( "file 2" );
}

// name with `%20` result in a `file 1` being opened
try ( Scanner scanner = new Scanner( new URL( "file:" + path1 ).openStream() ) ) {
	String data = scanner.nextLine();
	assertThat( data ).isEqualTo( "file 2" );
}

// name with a space ` ` result in a `file 1` being opened
try ( Scanner scanner = new Scanner( new URL( "file:" + path2 ).openStream() ) ) {
	String data = scanner.nextLine();
	assertThat( data ).isEqualTo( "file 2" );
}

// name with %20 result in a `file 1` being opened
try ( Scanner scanner = new Scanner( new URI( "file:" + path1 ).toURL().openStream() ) ) {
	String data = scanner.nextLine();
	assertThat( data ).isEqualTo( "file 2" );
}

// name with a space ` ` result in an exception
assertThatThrownBy( () -> {
		try ( Scanner scanner = new Scanner( new URI( "file:" + path2 ).toURL().openStream() ) ) {
			String data = scanner.nextLine();
			assertThat( data ).isEqualTo( "file 2" );
		}
	}
).isInstanceOf( URISyntaxException.class );
```

Reply:
That seems like a good idea, especially when in JDK 20:
```java
URL url;
try {
    @SuppressWarnings("deprecation")
    var _unused = url = new URL(base, ParseUtil.encodePath(name, false));
} catch (MalformedURLException e) {
    return null;
}
```
🙈 😃

Reply:
Created https://hibernate.atlassian.net/browse/HSEARCH-4765 and added reference to it in TODOs for future selves' 😄